base/thread_pool.h

```diff
@@ -108,12 +109,17 @@ class thread_pool_impl : public thread_pool {
   task_id dequeue(task& t);
 
 public:
+  // Constructs a thread pool with no worker threads. Use `run_worker` to enter a thread into the thread pool.
+  thread_pool_impl();
```

Comment:
Why not `= default;` ?

Reply:
Don't want it to be inline, a few objects with non-trivial constructors in this class.